js/runner.go

```diff
@@ -54,8 +53,9 @@ type Runner struct {
 	RPSLimit       *rate.Limiter
 	RunTags        *metrics.TagSet
 
-	console   *console
-	setupData []byte
+	console    *console
+	setupData  []byte
+	BufferPool *lib.BufferPool
```

Comment:
Why did you add a `BufferPool` to the `js.Runner`?

Reply:
I added it to reuse same pool for all VUs. I suppose ´js.Runner´ instance is unique for a load test execution.

Reply:
Yes, but is that actually better for performance than a separate `BufferPool` per VU? In the original simplistic checks I did for https://github.com/grafana/k6/issues/794, a global `sync.Pool` actually performed a lot worse than a per-VU `sync.Pool`. 

Maybe that has changed in recent Go and k6 versions, but we'd need to see a new benchmark comparison between both to justify the decision to go with a global pool. And, as a small nitpick, even if the the global pool is performing better now, it should still probably be unexported.

Reply:
I ran the test ([BenchmarkReadResponseBody](https://github.com/grafana/k6/blob/163ceb55a05ee9e8878cec81249bdf775cb21e97/cmd/tests/cmd_run_test.go#L1648)) with both configurations and to have a global pool improves memory and time execution. First execution (with global pool) allocate less memory and can iterate more times.

I think the result make sense because we reuse more the pool objects, so we don't have to allocate memory neither spend time creating new objects.

`
Global pool for all VUs

BenchmarkReadResponseBody-10               1    30455673750 ns/op   138908381552 B/op   69924811 allocs/op


     checks.........................: 100.00% ✓ 222192      ✗ 0     
     data_received..................: 15 GB   507 MB/s
     data_sent......................: 21 MB   689 kB/s
     http_req_blocked...............: avg=16.18µs  min=0s       med=1µs      max=75.67ms  p(90)=2µs      p(95)=3µs    
     http_req_connecting............: avg=5.29µs   min=0s       med=0s       max=31.02ms  p(90)=0s       p(95)=0s     
     http_req_duration..............: avg=2.31ms   min=74µs     med=1.14ms   max=93.67ms  p(90)=5.53ms   p(95)=8.32ms 
       { expected_response:true }...: avg=2.31ms   min=74µs     med=1.14ms   max=93.67ms  p(90)=5.53ms   p(95)=8.32ms 
     http_req_failed................: 0.00%   ✓ 0           ✗ 222192
     http_req_receiving.............: avg=420.75µs min=9µs      med=34µs     max=56.95ms  p(90)=938µs    p(95)=1.88ms 
     http_req_sending...............: avg=27.8µs   min=1µs      med=3µs      max=69.87ms  p(90)=15µs     p(95)=43µs   
     http_req_tls_handshaking.......: avg=0s       min=0s       med=0s       max=0s       p(90)=0s       p(95)=0s     
     http_req_waiting...............: avg=1.86ms   min=60µs     med=860µs    max=93.54ms  p(90)=4.62ms   p(95)=6.87ms 
     http_reqs......................: 222192  7378.159847/s
     iteration_duration.............: avg=108.23ms min=101.33ms med=105.72ms max=204.63ms p(90)=117.13ms p(95)=121.4ms
     iterations.....................: 27774   922.269981/s
     vus............................: 100     min=100       max=100 
     vus_max........................: 100     min=100       max=100 
     
Individual pool for VUs

BenchmarkReadResponseBody-10               1    30450333375 ns/op   174401310944 B/op   73585506 allocs/op

     checks.........................: 100.00% ✓ 217008      ✗ 0     
     data_received..................: 15 GB   495 MB/s
     data_sent......................: 20 MB   673 kB/s
     http_req_blocked...............: avg=27µs     min=0s       med=1µs      max=44.97ms  p(90)=2µs     p(95)=3µs     
     http_req_connecting............: avg=14.71µs  min=0s       med=0s       max=42.62ms  p(90)=0s      p(95)=0s      
     http_req_duration..............: avg=2.99ms   min=75µs     med=1.36ms   max=106.88ms p(90)=7.15ms  p(95)=10.76ms 
       { expected_response:true }...: avg=2.99ms   min=75µs     med=1.36ms   max=106.88ms p(90)=7.15ms  p(95)=10.76ms 
     http_req_failed................: 0.00%   ✓ 0           ✗ 217008
     http_req_receiving.............: avg=590.9µs  min=10µs     med=59µs     max=71.81ms  p(90)=1.14ms  p(95)=2.56ms  
     http_req_sending...............: avg=34.37µs  min=1µs      med=3µs      max=64.95ms  p(90)=15µs    p(95)=45µs    
     http_req_tls_handshaking.......: avg=0s       min=0s       med=0s       max=0s       p(90)=0s      p(95)=0s      
     http_req_waiting...............: avg=2.37ms   min=61µs     med=1.02ms   max=92.94ms  p(90)=5.8ms   p(95)=8.92ms  
     http_reqs......................: 217008  7209.308715/s
     iteration_duration.............: avg=110.64ms min=101.37ms med=107.07ms max=227.74ms p(90)=121.7ms p(95)=129.85ms
     iterations.....................: 27126   901.163589/s
     vus............................: 100     min=100       max=100 
     vus_max........................: 100     min=100       max=100 
`